streams/src/main/java/org/apache/kafka/streams/processor/TopologyDescription.java

```diff
@@ -0,0 +1,412 @@
+        }
+
+        @Override
+        public Set<Node> predecessors() {
+            return Collections.emptySet();
+        }
+
+        @Override
+        public Set<Node> successors() {
+            return successors;
+        }
+
+        @Override
+        public String toString() {
+            return "Source: " + name + "(topics: " + topics + ") --> " + nodeNames(successors);
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            final Source source = (Source) o;
+            // oit successor to avoid infinite loops
+            return name.equals(source.name)
+                && topics.equals(source.topics);
+        }
+
+        @Override
+        public int hashCode() {
+            int result = name.hashCode();
+            result = 31 * result + topics.hashCode();
+            // omit successor to avoid infinite loops
+            return result;
+        }
+    }
+
+    public final static class Processor implements Node {
+        /** The name of this processor node. */
+        final String name;
+        /** The names of all connected stores. */
+        public final Set<String> stores;
+        /** All predecessors of this processor node. */
+        final Set<Node> predecessors = new HashSet<>();
+        /** All successors of this processor node. */
+        final Set<Node> successors = new HashSet<>();
+
+        Processor(final String name,
+                  final Set<String> stores) {
+            this.name = name;
+            this.stores = stores;
+        }
+
+        @Override
+        public String name() {
+            return name;
+        }
+
+        @Override
+        public Set<Node> predecessors() {
+            return predecessors;
+        }
+
+        @Override
+        public Set<Node> successors() {
+            return successors;
+        }
+
+        @Override
+        public String toString() {
+            return "Processor: " + name + "(stores: " + stores + ") --> " + nodeNames(successors) + " <-- " + nodeNames(predecessors);
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            final Processor processor = (Processor) o;
+            // oit successor to avoid infinite loops
+            return name.equals(processor.name)
+                && stores.equals(processor.stores)
+                && predecessors.equals(processor.predecessors);
+        }
+
+        @Override
+        public int hashCode() {
+            int result = name.hashCode();
+            result = 31 * result + stores.hashCode();
+            result = 31 * result + predecessors.hashCode();
```

Comment:
This looks odd.  Why the same prime?

Also, as a general comment, why don't we use [Objects#hash()](https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#hash(java.lang.Object...)) instead of manually doing this?